core/templates/domain/exploration/exploration-id-validation.service.ts

```diff
@@ -16,19 +16,30 @@
  * @fileoverview Service to validate exploration id
  */
 
-require('domain/summary/exploration-summary-backend-api.service.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('ExplorationIdValidationService', [
-  'ExplorationSummaryBackendApiService',
-  function(ExplorationSummaryBackendApiService) {
-    return {
-      isExpPublished: function(explorationId) {
-        return ExplorationSummaryBackendApiService.
-          loadPublicExplorationSummaries([explorationId]).then(
-            function(summaries) {
-              return (summaries.length === 1 && summaries[0] !== null);
-            });
-      }
-    };
+import { ExplorationSummaryBackendApiService } from
+  'domain/summary/exploration-summary-backend-api.service.ts';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class ExplorationIdValidationService {
+  constructor(
+    private explorationSummartBackendApiService:
+      ExplorationSummaryBackendApiService) {}
+
+  isExpPublished(explorationId: string): Promise<boolean> {
+    return this.explorationSummartBackendApiService.
+      loadPublicExplorationSummaries([explorationId]).then(
+        function(summaries: string[]) {
```

Comment:
This is incorrect. Summaries are not a list of strings. Please determine the correct type by looking through the code and use it here.

Reply:
You might need to create an interface for the Summary type if it does not already exist and use that here instead of `string[]`.

Reply:
@kevintab95 I have made a fix for the interface for summaries. Can you please guide me as what is the correct standards for making a interface and where should I define the Interface?